.azure/pipelines/azure-pipelines-mirror-within-azdo.yml

```diff
@@ -0,0 +1,64 @@
+# Configure which branches trigger builds
+trigger:
+  batch: true
+  branches:
+    include:
+    - internal/release/6.0
+
+variables:
+  - group: Mirror-Credentials
+
+# Merges code from one AzDO branch into another
+jobs:
+- template: /eng/common/templates/jobs/jobs.yml
+  parameters:
+    enableTelemetry: true
+    helixRepo: dotnet/arcade
+    jobs:
+    - job: Merge_Azure_DevOps_Branches
+      enableSBOM: false
+      pool:
+        name: NetCore1ESPool-Internal
+        demands: ImageOverride -equals Build.Server.Amd64.VS2019
+      variables:
+      - name: WorkingDirectoryName
+        value: repo-dir
+      - name: AzdoRepo
+        value: https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-aspnetcore
+      - name: TargetBranchName
+        value: $(Build.SourceBranch)-nonstable
+      - name: BranchToMirror
+        value: $(Build.SourceBranch)
+      steps:
+      - script: |
+          git clone https://dn-bot:$(dn-bot-dnceng-build-rw-code-rw)@dev.azure.com/dnceng/internal/_git/$(AzdoRepo) $(WorkingDirectoryName)
+        displayName: Clone AzDO repo
+      - script: |
+          git checkout -b $(TargetBranchName) -t origin/$(TargetBranchName)
```

Comment:
nit: Can combine this step using `--branch $(TargetBranchName)` in the `clone` command